geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/QueryPlanner.scala

```diff
@@ -216,17 +223,25 @@ object QueryPlanner extends LazyLogging {
       // clear any configured hints so we don't process them again
       threadedHints.clear()
     }
+    // handle any params passed in through geoserver
+    QueryPlanner.handleGeoServerParams(query)
     // set transformations in the query
     QueryPlanner.setQueryTransforms(query, sft)
     // set return SFT in the query
     QueryPlanner.setReturnSft(query, sft)
-    // handle any params passed in through geoserver
-    QueryPlanner.handleGeoServerParams(query)
+  }
 
+  /**
+   * Return a new query with updated filters, ready to execute
+   *
+   * @param base query to update
+   * @param sft simple feature type to be queried
+   * @return new query with filters updated
+   */
+  def updateFilter(base: Query, sft: SimpleFeatureType): Query = {
+    val query = new Query(base)
     // add the bbox from the density query to the filter
-    // if configure has been called already, don't re-add - sometimes the bbox is split by
-    // IDL handling, causing us to not detect it as a duplicate
-    if (query.getHints.isDensityQuery && !query.getHints.isConfigured) {
+    if (query.getHints.isDensityQuery) {
```

Comment:
Yeah - I split the configure query stuff into two methods - the old configureQuery which can get called multiple times but is now idempotent, and then this method updateFilter, which will only get called once in the query planner just before calculating ranges, etc. It doesn't modify the existing query, it returns a new query - so technically the comment would still be valid, but it would never get called like that anymore. I'll add a comment to the method signature to make it clearer.